x-pack/legacy/plugins/alerting_ui/np_ready/public/application/sections/actions_list/components/actions_list.tsx

```diff
@@ -0,0 +1,264 @@
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { Action, ActionType, deleteActions, loadActions, loadActionTypes } from '../../../lib/api';
+import { ActionsContext } from '../../../context/app_context';
+import { useAppDependencies } from '../../../index';
+import { AlertingActionsDropdown } from './create_menu_popover';
+
+type ActionTypeIndex = Record<string, ActionType>;
+interface Pagination {
+  index: number;
+  size: number;
+}
+interface Data extends Action {
+  actionType: ActionType['name'];
+}
+
+export const ActionsList: React.FunctionComponent = () => {
+  const {
+    core: { http },
+    plugins: { capabilities, toastNotifications },
+  } = useAppDependencies();
+  const canDelete = capabilities.get().actions.delete;
+
+  const [actionTypesIndex, setActionTypesIndex] = useState<ActionTypeIndex | undefined>(undefined);
+  const [actions, setActions] = useState<Action[]>([]);
+  const [data, setData] = useState<Data[]>([]);
+  const [selectedItems, setSelectedItems] = useState<Data[]>([]);
+  const [isLoadingActionTypes, setIsLoadingActionTypes] = useState<boolean>(false);
+  const [isLoadingActions, setIsLoadingActions] = useState<boolean>(false);
+  const [isDeletingActions, setIsDeletingActions] = useState<boolean>(false);
+  const [totalItemCount, setTotalItemCount] = useState<number>(0);
+  const [page, setPage] = useState<Pagination>({ index: 0, size: 10 });
+  const [searchText, setSearchText] = useState<string | undefined>(undefined);
+
+  useEffect(() => {
+    loadActionsTable();
+  }, [page, searchText]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        setIsLoadingActionTypes(true);
+        const actionTypes = await loadActionTypes({ http });
+        const index: ActionTypeIndex = {};
+        for (const actionType of actionTypes) {
+          index[actionType.id] = actionType;
+        }
+        setActionTypesIndex(index);
+      } catch (e) {
+        toastNotifications.addDanger({
+          title: i18n.translate(
+            'xpack.alertingUI.sections.actionsList.unableToLoadActionTypesMessage',
+            { defaultMessage: 'Unable to load action types' }
+          ),
+        });
+      } finally {
+        setIsLoadingActionTypes(false);
+      }
+    })();
+  }, []);
+
+  useEffect(() => {
+    // Avoid flickering before action types load
+    if (typeof actionTypesIndex === 'undefined') {
+      return;
+    }
+    const updatedData = actions.map(action => {
+      return {
+        ...action,
+        actionType: actionTypesIndex[action.actionTypeId]
+          ? actionTypesIndex[action.actionTypeId].name
+          : action.actionTypeId,
+      };
+    });
+    setData(updatedData);
+  }, [actions, actionTypesIndex]);
+
+  async function loadActionsTable() {
+    setIsLoadingActions(true);
+    try {
+      const actionsResponse = await loadActions({ http, page, searchText });
```

Comment:
We might want to consider just loading all the actions and keeping them in memory.  This would make the UI more responsive.  I can't imagine that there would be more than a few hundred actions at most.

Reply:
Good idea, I went ahead and made the change here: https://github.com/elastic/kibana/pull/47843/commits/afffcf36905e7b34ebfa5188a367e19a5896541d.